packages/mask/src/components/shared/WalletStatusBox/TransactionList.tsx

```diff
@@ -0,0 +1,197 @@
+    },
+    cell: {
+        fontSize: 14,
+        display: 'flex',
+        alignItems: 'center',
+        padding: theme.spacing(0, 0.5),
+        color: theme.palette.text.primary,
+        boxSizing: 'border-box',
+    },
+    link: {
+        fill: 'none',
+    },
+    linkIcon: {
+        fill: 'none',
+        width: 12,
+        height: 12,
+        marginLeft: theme.spacing(0.5),
+    },
+}))
+
+const statusTextColorMap: Record<TransactionStatusType, string> = {
+    [TransactionStatusType.NOT_DEPEND]: '#FFB915',
+    [TransactionStatusType.SUCCEED]: '#60DFAB',
+    [TransactionStatusType.FAILED]: '#FF5F5F',
+    [TransactionStatusType.CANCELLED]: '#FF5F5F',
+}
+const getContractFunctionName = async (data: string | undefined) => {
+    if (!data) return null
+    const sig = data.slice(0, 10)
+    const name = await Services.Ethereum.getContractFunctionName(sig)
+    return name ? name.replace(/_/g, ' ') : 'Contract Interaction'
+}
+
+interface TransactionProps extends GridProps {
+    chainId: ChainId
+    transaction: RecentTransaction
+    onClear?(tx: RecentTransaction): void
+}
+const Transaction: FC<TransactionProps> = ({ chainId, transaction: tx, onClear = noop, ...rest }) => {
+    const { t } = useI18N()
+    const { Utils } = useWeb3State()
+    const { classes, theme } = useStyles()
+
+    const statusTextMap: Record<TransactionStatusType, string> = {
+        [TransactionStatusType.NOT_DEPEND]: t('recent_transaction_pending'),
+        [TransactionStatusType.SUCCEED]: t('recent_transaction_success'),
+        [TransactionStatusType.FAILED]: t('recent_transaction_failed'),
+        [TransactionStatusType.CANCELLED]: t('recent_transaction_cancelled'),
+    }
+
+    const web3 = useWeb3()
+    const { value: targetAddress } = useAsync(async () => {
+        if (tx.receipt?.contractAddress) return tx.receipt.contractAddress
+        if (tx.payload?.params?.[0].to) return tx.payload.params[0].to as string
+        const transaction = await web3.eth.getTransaction(tx.hash)
+        return transaction.to
+    }, [web3, tx])
+    const address = (targetAddress || '').toLowerCase()
+    const txData = tx.payload?.params?.[0].data as string | undefined
+    const { value: functionName } = useAsync(async () => {
+        return getContractFunctionName(txData)
+    }, [txData])
+
+    const handleClear = useCallback(() => {
+        onClear(tx)
+    }, [onClear, tx])
+
+    const domain = useMemo(() => getContractOwnerDomain(address), [address])
+
+    const [txStatus, setTxStatus] = useState(tx.status)
+
+    useEffect(() => {
+        return WalletMessages.events.transactionStateUpdated.on((data) => {
+            switch (data.type) {
+                case TransactionStateType.CONFIRMED:
+                    if (tx.hash === data.receipt.transactionHash) {
+                        setTxStatus(TransactionStatusType.SUCCEED)
+                    }
+                    break
+                case TransactionStateType.FAILED:
```

Comment:
you're missing `break`, and look like the `if(tx.hash === data.receipt.transactionHash)` move outside the `switch` be better.

Reply:
> you're missing break

We can omit it at the final branch
> look like the if(tx.hash === data.receipt.transactionHash) move outside the switch be better.

When a TX `receipt`, it also has receipt.hash
```ts
    | {
          type: TransactionStateType.RECEIPT
          receipt: TransactionReceipt
      }
    | {
          type: TransactionStateType.CONFIRMED
          no: number
          receipt: TransactionReceipt
          reason?: string
      }
```